src/IntrType.f90

```diff
@@ -440,6 +567,78 @@ ELEMENTAL FUNCTION approxeq_rel_double(a,b) RESULT(bool)
 ENDFUNCTION approxeq_rel_double
 !
 !-------------------------------------------------------------------------------
+!> @brief Defines the operation for performing a relative comparison of one
+!> single-precision and one double-precision reals with .APPROXEQR.
+!> @param a
+!> @param b
+!> @returns bool logical indicating if a and b are approximately equal
+!>
+!> This performs a relative comparison by scaling the default epsilon value to
+!> the size of the larger of the two. It should be used when @c and @b are of
+!> the same magnitude and very large or very small. If either @c a or @c b is
+!> zero (exactly) then this routine is equivalent to an absolute comparison.
+!>
+ELEMENTAL FUNCTION approxeq_rel_mixsd(a,b) RESULT(bool)
```

Comment:
I would suggest you downscale and then call the existing routine for this

Reply:
Just too many of them in the unit tests which makes the downscale inconvenient (and increase the length and reduce readability of the tests).

Reply:
I'm not suggesting to downscale in the unit test, but to downscale here.  This way you aren't duplicating code logic.:
```
ELEMENTAL FUNCTION approxeq_rel_mixsd(a,b) RESULT(bool)
  REAL(SSK),INTENT(IN) :: a
  REAL(SDK),INTENT(IN) :: b
  LOGICAL(SBK) :: bool
  bool=approxeq_rel_single(a, REAL(b,SSK))
ENDFUNCTION approxeq_rel_mixsd
```

Reply:
Ah... makes sense.